stock_picking_declared_value/views/stock_move_views.xml

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2025 Nitrokey GmbH
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <!-- Add declared value fields to the stock move form view -->
+    <record id="view_move_form_inherit_declared_value" model="ir.ui.view">
+        <field name="name">stock.move.form.inherit.declared.value</field>
+        <field name="model">stock.move</field>
+        <field name="inherit_id" ref="stock.view_move_form" />
+        <field name="arch" type="xml">
+            <field name="product_uom" position="after">
+                <field name="currency_id" invisible="1" />
+                <field name="sale_line_id" invisible="1" />
+                <field
+                    name="sale_unit_price"
+                    attrs="{'invisible': [('sale_line_id', '=', False)]}"
```

Comment:
But you still set the values if there is no `sale_line_id`. You would hide these with the invisible here.